### PR TITLE
add nightly EventStore build to CI

### PR DESCRIPTION
**Pull request by the-mikedavis**
connects #60

adds the image that the eventstore team publishes nightly to the CI matrix

useful for sniffing out new protobuf breakages and testing features not yet released

this also allows us to enable some tests for upcoming unreleased features. one of those tests is now enabled with

```elixir
@tag compatible(:nightly)
```

### CODE REVIEW

File: test/spear_test.exs [hunk 1]
```diff
@@ -919,28 +919,15 @@ defmodule SpearTest do
       assert Spear.stream!(c.conn, c.stream_name) |> Enum.to_list() == []
     end
 
-    # deadlines do not appear to be working no matter what timestamp I put on
-    # or how I interleave the messages
-    # @tag compatible("~> 21.6")
-    @tag :skip
+    @tag compatible(:nightly)
```
Comment (the-mikedavis) [inline test/spear_test.exs:922]: fixed https://github.com/EventStore/EventStore/pull/3138

nice to be able to test it now in CI and see it pass :rocket:

File: test/spear_test.exs [hunk 2]
```diff
@@ -919,28 +919,15 @@ defmodule SpearTest do
...
-      assert {:ok, ^batch_id, ^request_id} =
-               random_events()
-               |> Stream.drop(5)
-               |> Stream.take(5)
-               |> Spear.append_batch(c.conn, request_id, c.stream_name,
-                 deadline: deadline,
-                 done?: true,
-                 batch_id: batch_id
-               )
+               |> Spear.append_batch(c.conn, :new, c.stream_name, deadline: deadline)
```
Comment (the-mikedavis) [inline test/spear_test.exs:930]: the red lines were me fiddling around trying to make this fail before I realized it was a bug

**Review by tonyvanriet — APPROVED** [review]
cool

**Review by bjornrud — COMMENTED** [review]
This is good lazy.

poodle intently listening and note taking

**Pull request merged**